### PR TITLE
Sharing namespaces

### PR DESCRIPTION
**Pull request by crufter**
Caveats: invite create must be opened up to be public, custom auth happens in create.

The test fails because it needs changes from the micro PR.

### CODE REVIEW

File: invite/handler/handler.go [hunk 1]
```diff
@@ -30,8 +32,17 @@ type Invite struct {
 
 // Create an invite
 func (h *Invite) Create(ctx context.Context, req *pb.CreateRequest, rsp *pb.CreateResponse) error {
+	account, ok := auth.AccountFromContext(ctx)
+	if !ok {
+		return errors.BadRequest(h.name, "Unauthorized request")
```
Comment (ben-toogood) [inline invite/handler/handler.go:37]: Unauthorized not BadReq

File: invite/proto/invite.proto [hunk 2]
```diff
@@ -13,6 +13,7 @@ service Invite {
 
 message CreateRequest {
   string email = 1;
+  string namespace = 2;
```
Comment (ben-toogood) [inline invite/proto/invite.proto:16]: This isn't used?

Comment (crufter) [reply invite/proto/invite.proto:16]: This will be used early in the flow in the CLI.

File: go.mod [hunk 3]
```diff
@@ -16,3 +17,7 @@ require (
 )
 
 replace google.golang.org/grpc => google.golang.org/grpc v1.26.0
+
+replace github.com/micro/micro/v3 => github.com/micro/micro/v3 v3.0.0-beta.0.20200821123950-a4768072bebb
```
Comment (crufter) [inline go.mod:21]: Will remove these replaces obv.

File: signup/handler/signup.go [hunk 4]
```diff
@@ -327,14 +347,22 @@ func (e *Signup) CompleteSignup(ctx context.Context, req *signup.CompleteSignupR
 		secret = uuid.New().String()
 	}
 
-	ns, err := e.createNamespace(ctx)
-	if err != nil {
-		return err
+	ns := ""
+	fmt.Println("namespaces", namespaces)
```
Comment (crufter) [inline signup/handler/signup.go:351]: will remove

Comment (domwong) [inline signup/handler/signup.go:351]: delete

**Review by domwong — APPROVED** [review]
LGTM

File: signup/handler/signup.go [hunk 5]
```diff
@@ -299,51 +345,96 @@ func (e *Signup) CompleteSignup(ctx context.Context, req *signup.CompleteSignupR
...
+			return merrors.InternalServerError("signup", "Error listing subscriptions: %v", err)
+		}
+		var addUserPlan *paymentsproto.Plan
+		for _, sub := range subs.Subscriptions {
+			if sub.Plan.Id == e.additionUsersPriceID {
+				addUserPlan = sub.Plan
+			}
+		}
+		if addUserPlan == nil {
+			logger.Info("reating with quantity 1")
```
Comment (domwong) [inline signup/handler/signup.go:367]:
```suggestion
			logger.Info("Creating with quantity 1")
```

**Review by domwong — APPROVED** [review]

File: signup/handler/signup.go (hunk 5 above)
Comment (crufter) [reply signup/handler/signup.go]: Done

**Pull request merged**